app/models/allocation.rb

```diff
@@ -53,15 +53,17 @@ class Allocation < VersionedModel
   after_initialize :initialize_state
   delegate :fill, :unfill, to: :state_machine
 
-  def cancel
-    comment = 'Allocation was cancelled'
-
+  def cancel(cancel_moves: true, reason: CANCELLATION_REASON_OTHER, comment: 'Allocation was cancelled')
```

Comment:
This is pretty clunky now - but mostly temporary. With the new cancellation endpoint we don't actually want to cancel the moves here - instead we create events that trigger move cancellation and run those events in sequence. For the legacy endpoint we do want the original behaviour of cancelling moves here.

Also it's now possible to pass cancellation reason and comment as params (in the new endpoint) where previously (in the old endpoint) those were hardcoded values.